code/modules/mob/interactive.dm

```diff
@@ -0,0 +1,643 @@
+				src.say("[pick("Innocent","Guilty","Traitorous","Honk")] until proven [pick(adjective_generic)]!")
+			if(5)
+				var/toSay = ""
+				for(var/i = 0; i < 5; i++)
+					toSay += "[pick(curse_words)] "
+				src.say("Hey [pick(nouns_generic)], why dont you go [toSay], you [pick(nouns_insult)]!")
+
+/mob/living/carbon/human/interactive/proc/shitcurity(obj)
+	if(retal && TARGET)
+		for(var/obj/item/I in src.contents)
+			if(istype(I,/obj/item/weapon/restraints/handcuffs))
+				take_to_slot(I)
+				I.attack(TARGET,src)
+				sleep(25)
+
+/mob/living/carbon/human/interactive/proc/sidestep(obj)
+	var/shift = 0
+	for(var/dir in cardinal)
+		var/turf/T = get_step(src,dir)
+		if(T)
+			for(var/obj/A in T.contents)
+				if(denied_filter(A))
+					shift = 1
+			if(T.density)
+				shift = 1
+			if(shift)
+				if(src.dir == NORTH || SOUTH)
+					var/towalk = pick(EAST,WEST)
+					walk_to(src,get_step(src,towalk))
+				else
+					var/towalk = pick(NORTH,SOUTH)
+					walk_to(src,get_step(src,towalk))
+
+/mob/living/carbon/human/interactive/proc/combat(obj)
+	set background = 1
+	if(canmove)
+		if(prob(attitude) && (graytide || (TRAITS & TRAIT_MEAN)) || retal)
+			a_intent = "harm"
+			zone_sel.selecting = pick("chest","r_leg","l_leg","r_arm","l_arm","head")
+			doing |= FIGHTING
+			if(retal)
+				TARGET = retal_target
+			else
+				var/mob/living/M = locate(/mob/living) in oview(7,src)
+				if(istype(M,/mob/living/carbon/human/interactive/greytide))
+					return
+				if(M != src)
+					TARGET = M
+				if(!M)
+					doing = doing & ~FIGHTING
+
+	if((TARGET && (doing & FIGHTING)) || graytide) // this is a redundancy check
+		var/mob/living/M = TARGET
+		if(istype(M,/mob/living))
+			if(M in oview(14,src))
+				if(M.health > 10)
+					if(main_hand)
+						if(main_hand.force != 0)
+							if(istype(main_hand,/obj/item/weapon/gun/projectile))
+								var/obj/item/weapon/gun/projectile/P = main_hand
+								if(!P.chambered)
+									P.chamber_round()
+									P.update_icon()
+								else if(P.get_ammo(1) == 0)
+									P.attack_self(src)
+								else
+									if(prob(robustness))
+										P.afterattack(TARGET, src)
+							else
+								if(get_dist(src,TARGET) > 2)
+									if(!walk2derpless(TARGET))
+										timeout++
+								else
+									var/obj/item/weapon/W = main_hand
+									if(prob(robustness))
+										W.attack(TARGET,src)
+							sleep(1)
+					else
+						if(get_dist(src,TARGET) > 2)
+							tryWalk(TARGET)
+						else
+							if(Adjacent(TARGET))
+								M.attack_hand(src)
+								sleep(1)
+			else if(M.health <= 1 || !(M in oview(14,src)))
+				doing = doing & ~FIGHTING
+				timeout = 0
+				TARGET = null
+				retal = 0
+				retal_target = null
+		else
+			timeout++
+
+/mob/living/carbon/human/interactive/proc/doorscan(obj)
+	for(var/dir in cardinal)
+		var/turf/T = get_step(src,dir)
+		if(T)
+			for(var/obj/machinery/door/D in T.contents)
+				if(D.check_access(MYID) && !istype(D,/obj/machinery/door/poddoor) && D.layer == 3.1)

```

Comment:
Instead of this, why not simply check for density, which is what the door open() proc actually does.